packages/features/bookings/repositories/BookingRepository.ts

```diff
@@ -1383,4 +1383,86 @@ export class BookingRepository {
       },
     });
   }
+
+  findBookingForRequestReschedule({ bookingUid }: { bookingUid: string }) {
+    return this.prismaClient.booking.findUniqueOrThrow({
+      where: {
+        uid: bookingUid,
+        NOT: {
+          status: {
+            in: [BookingStatus.CANCELLED, BookingStatus.REJECTED],
+          },
+        },
+      },
+      select: {
+        id: true,
+        uid: true,
+        userId: true,
+        user: {
+          select: {
+            id: true,
+            email: true,
+          },
+        },
+        title: true,
+        description: true,
+        startTime: true,
+        endTime: true,
+        eventTypeId: true,
+        userPrimaryEmail: true,
+        eventType: {
+          select: {
+            teamId: true,
+            parentId: true,
+            slug: true,
+            hideOrganizerEmail: true,
+            customReplyToEmail: true,
+            bookingFields: true,
+            metadata: true,
+            team: {
+              select: {
+                id: true,
+                name: true,
+                parentId: true,
+              },
+            },
+          },
+        },
+        location: true,
+        attendees: true,
+        references: true,
+        customInputs: true,
+        dynamicEventSlugRef: true,
+        dynamicGroupSlugRef: true,
+        destinationCalendar: true,
+        smsReminderNumber: true,
+        workflowReminders: true,
+        responses: true,
+        iCalUID: true,
+      },
+    });
+  }
+
+  async markBookingAsRescheduled({
+    bookingId,
+    cancellationReason,
+    cancelledBy,
```

Comment:
Also, in the context here, we should name the variables as 
 _requestedRescheduleBy_ and _requestedRescheduleReason_